### ISSUE TITLE
Kind occurrence check deferred to the object definition.

### ISSUE THREAD

**Issue by Heimdell** [Frege/frege#258]
The next code compiles fine:
```frege
module hello.HelloWorld where

type Omega s = s s

data Cons a b = Cons a b

type Stream a = Omega (Cons a)

--infinite :: a -> Stream a
--infinite x = Cons x (infinite x)

data Const a b = Const { getConst :: a }
```
although, `Omega` type should clearly produce a "cyclic kind" error.

If we uncomment the `infinite` function, then compiler complains on `Stream` type.

GHC will yell on the `Omega` itself, which is the best behaviour for me.

**Comment by Ingo60** [comment]
Yes, currently the compiler only checks that the *type* definitions don't construct cycles. As long as you don't use them, you can write utter nonsense on the RHS.

**Comment by Ingo60** [comment]
To solve this, a more advanced kind inference is needed.
